mysql-test/mysql-test-run.pl

```diff
@@ -1719,17 +1720,28 @@ sub command_line_setup {
     $opt_valgrind= 1;
     $opt_valgrind_mysqld= 1;
 
+    my @valgrind_args_orig= @valgrind_args;
+
+    unshift(@valgrind_args, "--tool=callgrind");
     # Set special valgrind options unless options passed on command line
     push(@valgrind_args, "--trace-children=yes")
-      unless @valgrind_args;
+      unless @valgrind_args_orig;
   }
 
-  if ( $opt_valgrind )
+  # default to --tool=memcheck
+  if ($opt_valgrind && ! grep(/^--tool=/i, @valgrind_args))
   {
     # Set valgrind_options to default unless already defined
-    push(@valgrind_args, @default_valgrind_args)
+    push(@valgrind_args, @default_memcheck_valgrind_args)
       unless @valgrind_args;
+    push(@valgrind_args, "--suppressions=${glob_mysql_test_dir}/valgrind.supp")
+      if -f "$glob_mysql_test_dir/valgrind.supp";
```

Comment:
Currently we indeed only add suppressions along with memcheck, but is it even right? Shouldn't they always be there? According to valgrind documentation, the option is basic, applicable to all tools. Of course, now the file in MTR only contains Memcheck errors, but that's because only Memcheck has been used; if we want to use other tools as well, it might make sense to all them to suppress things too.